operator/controllers/syncer/csi_node.go

```diff
@@ -103,8 +103,9 @@ func (s *csiNodeSyncer) SyncCSIDaemonsetFn(daemonSetRestartedKey string, daemonS
 			secrets = append(secrets, corev1.LocalObjectReference{Name: s})
 		}
 	} else {
-		// Use default imagePullSecret
-		secrets = append(secrets, corev1.LocalObjectReference{Name: config.DefaultImagePullSecret})
+		// Use default imagePullSecrets
+		secrets = append(secrets, corev1.LocalObjectReference{Name: config.ImagePullSecretRegistryKey},
+			corev1.LocalObjectReference{Name: config.ImagePullSecretEntitlementKey})
```

Comment:
I think we need ibm-spectrum-scale-csi-registrykey,ibm-entitlement-key in any case even if user has defined the pull secret in the cr object. @deeghuge any input ?

Reply:
As discussed earlier, the current implementation of imagePullSecrets is such that: if user specifies imagePullSecrets from driver CR, we are not using default imagePullSecrets. Let me post a message on our slack channel to discuss if we want to change this behavior and close on this quickly.

Reply:
yes, both secret must be there always so else part can be removed

Reply:
secret always will be ImagePullSecretRegistryKey, ImagePullSecretEntitlementKey, <user defined secrets>

Reply:
Changed this behavior